components/ntp_sponsored_images/browser/ntp_sponsored_image_source.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_NTP_SPONSORED_IMAGES_BROWSER_NTP_SPONSORED_IMAGE_SOURCE_H_
+#define BRAVE_COMPONENTS_NTP_SPONSORED_IMAGES_BROWSER_NTP_SPONSORED_IMAGE_SOURCE_H_
+
+#include <string>
+
+#include "base/files/file_path.h"
+#include "base/memory/weak_ptr.h"
+#include "base/optional.h"
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_internal_data.h"
+#include "content/public/browser/url_data_source.h"
+
+class NTPSponsoredImagesComponentManager;
+
+// This serves branded image data.
+// This referes with weak ptr because both can have different life cycle.
+class NTPSponsoredImageSource : public content::URLDataSource {
+ public:
+  explicit NTPSponsoredImageSource(
+      const NTPSponsoredImagesInternalData& internal_images_data);
+
+  ~NTPSponsoredImageSource() override;
+
+  NTPSponsoredImageSource(const NTPSponsoredImageSource&) = delete;
```

Comment:
use `DISALLOW_COPY_AND_ASSIGN`

Reply:
OK to leave as-is, IMO. See note at https://github.com/brave/brave-core/pull/4075/files/2f7f51f8a0ffee135d823311f7bb464dd74e6acc#r370352211